enzy_htp/analysis/rmsd.py

```diff
@@ -0,0 +1,120 @@
+#! python3
+# -*- encoding: utf-8 -*-
+'''
+@File    :   rmsd.py
+@Created :   2024/07/12 18:45
+@Author  :   Zhong, Yinjie
+@Version :   2.0
+@Contact :   yinjie.zhong@vanderbilt.edu
+'''
+
+# Here put the import lib.
+from typing import List, Dict, Union
+
+# Here put EnzyHTP imports.
+from enzy_htp import interface as eh_interface, _LOGGER
+from enzy_htp import config as eh_config
+from enzy_htp._interface.pymol_interface import PyMolInterface
+from enzy_htp._interface.amber_interface import AmberInterface
+from enzy_htp.core import file_system as fs
+from enzy_htp.structure import Residue, Structure
+from enzy_htp.structure.structure_ensemble import StructureEnsemble
+
+ACCEPTED_INTERFACES: Dict[str, Union[PyMolInterface, AmberInterface]] = {
+    "pymol": eh_interface.pymol,
+    "amber": eh_interface.amber,
+}
+
+def compose_mask_pattern(mask_region: List[Residue] = list(), interface: str = "amber", ca_only: bool = True) -> str:
+    """Compose the mask pattern from the given mask region for Amber/CppTraj or PyMOL interface.
+    
+    Args:
+        mask_region (List[Residue], optional): A list of Residues to calculation RMSD value. If the list is empty, then all residues are selected.
+        interface (str, optional): The interface for calculation. Default amber.
+        include_ligand (bool, optional): Indicate if ligands are included in RMSD calculation. Default True.
+        ca_only (bool, optional): Indicate if only C-alpha are included in RMSD calculation; otherwise all atoms except hydrogens are included. Default True.
+    """
+    interface = interface.lower()
+    mask_pattern = str()
+    
+    if (interface == "amber"):
+        if (mask_region):
+            mask_pattern = f":{','.join([str(residue.idx) for residue in mask_region])}"
+        else:
+            mask_pattern = ""
+        if (ca_only):
+            if mask_pattern:
+                mask_pattern += "&"
+            mask_pattern += "@CA"
+            
+    elif (interface == "pymol"):
+        if (mask_region):
+            mask_pattern = f"resi {'+'.join([str(residue.idx) for residue in mask_region])}"
+        else:
+            mask_pattern = "all"
+        if (ca_only):
+            mask_pattern += " and name CA"
+
+    elif (interface not in ACCEPTED_INTERFACES.keys()):
+        raise ValueError(f"Unacceptable interface value. Please choose from {' and '.join(ACCEPTED_INTERFACES.keys())}")
+    return mask_pattern
+
```

Comment:
We need one single API for the RMSD calculation since it is getting the same type of property. (i.e.: having the same I/O in principle) Imagining having the current set up in the web app.

Reply:
Since you already of the many ones, just create a master API that handles the dispatch of the three. (dont need to use the plum.dispatch)